File: 0001-two-sum/0001-two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    const mp = new Map()
    for (let i = 0; i < nums.length; i++) {
        if (mp.has(nums[i])) {
            return[ mp.get(nums[i]) , i]
        }
        mp.set(target - nums[i], i)
    }
    return [0 , 0]
};

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    numberMap := make(map[int]int)

    for index, number := range nums {
        neededNumber := target - number
        if complementIndex, ok := numberMap[neededNumber]; ok {
            return []int{complementIndex, index}
        }
        numberMap[number] = index
    }

    return []int{}
}